web/src/components/organisms/Settings/Members/hooks.ts

```diff
@@ -0,0 +1,186 @@
+          : data?.me?.myWorkspace ?? undefined
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentWorkspace, setWorkspace, workspaces, data?.me]);
+
+  const [searchUserQuery, { data: searchUserData }] =
+    useGetUserBySearchLazyQuery();
+
+  useEffect(() => {
+    changeSearchedUser(searchUserData?.searchUser ?? undefined);
+  }, [searchUserData?.searchUser]);
+
+  const handleUserSearch = useCallback(
+    (nameOrEmail: string) =>
+      nameOrEmail && searchUserQuery({ variables: { nameOrEmail } }),
+    [searchUserQuery]
+  );
+
+  const [addMemberToWorkspaceMutation] = useAddMemberToWorkspaceMutation();
+
+  const handleMemberAddToWorkspace = useCallback(
+    async (userIds: string[]) => {
+      const results = await Promise.all(
+        userIds.map(async (userId) => {
+          if (!workspaceId) return;
+          const result = await addMemberToWorkspaceMutation({
+            variables: { userId, workspaceId, role: Role.Reader },
+            refetchQueries: ["GetWorkspaces"],
+          });
+          const workspace = result.data?.addMemberToWorkspace?.workspace;
+          if (result.errors || !workspace) {
+            // TODO: notification
+            return;
+          }
+          setWorkspace(workspace);
+        })
+      );
+      if (results) {
+        // TODO: notification
+      }
+    },
+    [workspaceId, addMemberToWorkspaceMutation, setWorkspace]
+  );
+
+  const [updateMemberOfWorkspaceMutation] =
+    useUpdateMemberOfWorkspaceMutation();
+
+  const handleMemberOfWorkspaceUpdate = useCallback(
+    async (userId: string, role: RoleUnion) => {
+      if (workspaceId) {
+        const results = await updateMemberOfWorkspaceMutation({
+          variables: {
+            workspaceId,
+            userId,
+            role: {
+              READER: Role.Reader,
+              WRITER: Role.Writer,
+              OWNER: Role.Owner,
+            }[role],
+          },
+        });
+        const workspace = results.data?.updateMemberOfWorkspace?.workspace;
+        if (workspace) {
+          setWorkspace(workspace);
+        }
+      }
+    },
+    [workspaceId, setWorkspace, updateMemberOfWorkspaceMutation]
+  );
+
+  const [removeMemberFromWorkspaceMutation] =
+    useRemoveMemberFromWorkspaceMutation();
+
+  const handleMemberRemoveFromWorkspace = useCallback(
+    async (userId: string) => {
+      if (!workspaceId) return;
+      const result = await removeMemberFromWorkspaceMutation({
+        variables: { workspaceId, userId },
+        refetchQueries: ["GetWorkspaces"],
+      });
+      const workspace = result.data?.removeMemberFromWorkspace?.workspace;
+      if (result.errors || !workspace) {
+        // TODO: notification
+        return;
+      }
+      setWorkspace(workspace);
+      // TODO: notification
+    },
+    [workspaceId, removeMemberFromWorkspaceMutation, setWorkspace]
+  );
+
+  const handleRoleModalClose = useCallback(
+    (r?: boolean) => {
```

Comment:
What is the `r` for? Is it for role, or refetch? Can you name it a little more meaningfully?